File: obra-social-g5-backend/src/main/java/org/softek/g5/security/usuario/Usuario.java
package org.softek.g5.security.usuario;

import io.quarkus.hibernate.orm.panache.PanacheEntityBase;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Entity
@Data
@Table(name = "usuario")
@EqualsAndHashCode(callSuper=false)
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Usuario extends PanacheEntityBase{

	@Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
	
	private String username;
	private String password;
	private String email;
	private UsuarioRolesEnum rol;
	private Boolean estaEliminado;
	private Boolean primerInicioSesion;
	
}

File: obra-social-g5-frontend/almedin-vite-ts/src/assets/models/Usuario.ts
export interface Usuario {
    id: number;
    username: string;
    password: string;
    email: string;
    rol: UsuarioRolesEnum;
    estaEliminado: boolean;
}
export enum UsuarioRolesEnum {
    ROL_SOCIO = 'ROL_SOCIO',
    ROL_ADMIN = 'ROL_ADMIN',
    ROL_RECEPCIONISTA = 'ROL_RECEPCIONISTA',
    ROL_MEDICO = 'ROL_MEDICO'
}
export interface UsuarioRequestDto {
    rol: UsuarioRolesEnum;
    username: string;
    password: string;
    email: string;
}

export interface UsuarioLoginDto {
    username: string;
    password: string;
}
export interface UsuarioResponseDto {
    id: number;
    rol: UsuarioRolesEnum;
    username: string;
    token: string;
}